index.js

```diff
@@ -82,35 +88,17 @@ function buildProcessTree (parentPid, tree, pidsToProcess, cb) {
             if (Object.keys(pidsToProcess).length == 0) {
                 cb();
             }
-            return
-        }
-
-        pids = [];
-        pid = '';
-        for (i = 0; i < allData.length; i++) {
-            if (allData[i] == '\n') {
-                pids.push(parseInt(pid, 10));
-                pid = '';
-                continue;
-            }
-            if (allData[i] != ' ') {
-                pid += allData[i];
-            }
+            return;
         }
 
-        pids.forEach(function (pid) {
-            tree[parentPid].push(pid)
-            tree[pid] = [];
-            pidsToProcess[pid] = 1;
-            buildProcessTree(pid, tree, pidsToProcess, cb);
+        allData.match(/\d+/g).forEach(function (pid) {
+          pid = parseInt(pid, 10);
```

Comment:
Slightly concerned, because this is not semantically equivalent to the old for loop. This only matches runs of digits, while the old for loop handled anything that wasn't a space ' '. But that said, the new code is "more correct" because pids should only be integer numbers anyway.